File: wmcW/src/components/Modals.js
import React from 'react';

const modalStyles = {
  overlay: {
    position: "fixed",
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    backgroundColor: "rgba(0, 0, 0, 0.5)",
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    zIndex: 1000,
    padding: "20px"
  },
  content: {
    backgroundColor: "white",
    borderRadius: "16px",
    padding: "30px",
    maxWidth: "500px",
    width: "90%",
    maxHeight: "80vh",
    overflowY: "auto",
    boxShadow: "0 20px 25px -5px rgba(0, 0, 0, 0.1)",
    position: "relative"
  },
  closeButton: {
    position: "absolute",
    top: "15px",
    right: "15px",
    background: "none",
    border: "none",
    fontSize: "24px",
    cursor: "pointer",
    color: "#64748b",
    width: "30px",
    height: "30px",
    borderRadius: "50%",
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    transition: "all 0.3s ease",
    zIndex: 10
  },
  title: {
    fontSize: "24px",
    fontWeight: "bold",
    marginBottom: "20px",
    color: "#1e293b",
    paddingRight: "40px"
  },
  inputGroup: {
    marginBottom: "20px"
  },
  label: {
    display: "block",
    marginBottom: "8px",
    fontWeight: "600",
    color: "#374151"
  },
  input: {
    width: "100%",
    padding: "12px 16px",
    border: "2px solid #d1d5db",
    borderRadius: "8px",
    fontSize: "16px",
    transition: "all 0.3s ease",
    fontFamily: "Arial, sans-serif"
  },
  textarea: {
    width: "100%",
    padding: "12px 16px",
    border: "2px solid #d1d5db",
    borderRadius: "8px",
    fontSize: "16px",
    minHeight: "100px",
    resize: "vertical",
    fontFamily: "Arial, sans-serif"
  },
  actions: {
    display: "flex",
    gap: "15px",
    justifyContent: "flex-end",
    marginTop: "25px"
  },
  cancelButton: {
    backgroundColor: "#6b7280",
    color: "white",
    border: "none",
    padding: "12px 24px",
    borderRadius: "8px",
    cursor: "pointer",
    fontSize: "14px",
    fontWeight: "600",
    transition: "all 0.3s ease"
  },
  confirmButton: {
    backgroundColor: "#2563eb",
    color: "white",
    border: "none",
    padding: "12px 24px",
    borderRadius: "8px",
    cursor: "pointer",
    fontSize: "14px",
    fontWeight: "600",
    transition: "all 0.3s ease"
  }
};

export const NewCourseModal = ({ viewModel, onUpdate }) => {
  const handleCreate = () => {
    const result = viewModel.createCourse();
    if (result) {
      onUpdate();
    }
  };

  const handleClose = () => {
    viewModel.setShowNewCourseModal(false);
    onUpdate();
  };

  return (
    <div style={modalStyles.overlay} onClick={handleClose}>
      <div style={modalStyles.content} onClick={e => e.stopPropagation()}>
        <button 
          style={modalStyles.closeButton}
          onClick={handleClose}
          onMouseEnter={(e) => e.target.style.backgroundColor = "#f3f4f6"}
          onMouseLeave={(e) => e.target.style.backgroundColor = "transparent"}
        >
          ×
        </button>
        
        <h2 style={modalStyles.title}>Create New Course</h2>
        
        <div style={modalStyles.inputGroup}>
          <label style={modalStyles.label}>Course Title *</label>
          <input 
            style={modalStyles.input} 
            placeholder="Ex: Advanced Python Programming" 
            value={viewModel.newCourse.title}
            onChange={(e) => {
              viewModel.updateNewCourse('title', e.target.value);
              onUpdate();
            }}
          />
        </div>
        
        <div style={modalStyles.inputGroup}>
          <label style={modalStyles.label}>Category *</label>
          <input 
            style={modalStyles.input} 
            placeholder="Ex: Web Development, Data Science" 
            value={viewModel.newCourse.category}
            onChange={(e) => {
              viewModel.updateNewCourse('category', e.target.value);
              onUpdate();
            }}
          />
        </div>
        
        <div style={modalStyles.inputGroup}>
          <label style={modalStyles.label}>Description</label>
          <textarea 
            style={modalStyles.textarea} 
            placeholder="Describe what students will learn in this course..." 
            value={viewModel.newCourse.description}
            onChange={(e) => {
              viewModel.updateNewCourse('description', e.target.value);
              onUpdate();
            }}
          />
        </div>
        
        <div style={modalStyles.actions}>
          <button 
            style={modalStyles.cancelButton}
            onClick={handleClose}
          >
            Cancel
          </button>
          <button 
            style={{
              ...modalStyles.confirmButton,
              opacity: (!viewModel.newCourse.title || !viewModel.newCourse.category) ? 0.6 : 1,
              cursor: (!viewModel.newCourse.title || !viewModel.newCourse.category) ? "not-allowed" : "pointer"
            }}
            onClick={handleCreate}
            disabled={!viewModel.newCourse.title || !viewModel.newCourse.category}
          >
            Create Course
          </button>
        </div>
      </div>
    </div>
  );
};

export const NewLiveModal = ({ viewModel, onUpdate }) => {
  const handleCreate = () => {
    const result = viewModel.createLiveSession();
    if (result) {
      onUpdate();
    }
  };

  const handleClose = () => {
    viewModel.setShowNewLiveModal(false);
    onUpdate();
  };

  return (
    <div style={modalStyles.overlay} onClick={handleClose}>
      <div style={modalStyles.content} onClick={e => e.stopPropagation()}>
        <button 
          style={modalStyles.closeButton}
          onClick={handleClose}
          onMouseEnter={(e) => e.target.style.backgroundColor = "#f3f4f6"}
          onMouseLeave={(e) => e.target.style.backgroundColor = "transparent"}
        >
          ×
        </button>
        
        <h2 style={modalStyles.title}>Schedule Live Session</h2>
        
        <div style={modalStyles.inputGroup}>
          <label style={modalStyles.label}>Session Title *</label>
          <input 
            style={modalStyles.input} 
            placeholder="Ex: Python Q&A Session" 
            value={viewModel.newLive.title}
            onChange={(e) => {
              viewModel.updateNewLive('title', e.target.value);
              onUpdate();
            }}
          />
        </div>
        
        <div style={modalStyles.inputGroup}>
          <label style={modalStyles.label}>Date & Time *</label>
          <input 
            style={modalStyles.input} 
            type="datetime-local" 
            value={viewModel.newLive.datetime}
            onChange={(e) => {
              viewModel.updateNewLive('datetime', e.target.value);
              onUpdate();
            }}
          />
        </div>
        
        <div style={modalStyles.inputGroup}>
          <label style={modalStyles.label}>Description</label>
          <textarea 
            style={modalStyles.textarea} 
            placeholder="Describe what you'll cover in this live session..." 
            value={viewModel.newLive.description}
            onChange={(e) => {
              viewModel.updateNewLive('description', e.target.value);
              onUpdate();
            }}
          />
        </div>
        
        <div style={modalStyles.actions}>
          <button 
            style={modalStyles.cancelButton}
            onClick={handleClose}
          >
            Cancel
          </button>
          <button 
            style={{
              ...modalStyles.confirmButton,
              opacity: (!viewModel.newLive.title || !viewModel.newLive.datetime) ? 0.6 : 1,
              cursor: (!viewModel.newLive.title || !viewModel.newLive.datetime) ? "not-allowed" : "pointer"
            }}
            onClick={handleCreate}
            disabled={!viewModel.newLive.title || !viewModel.newLive.datetime}
          >
            Schedule Session
          </button>
        </div>
      </div>
    </div>
  );
};

File: wmcm/lib/views/components/modals.dart
import 'package:flutter/material.dart';
import '../../viewmodels/dashboard_viewmodel.dart';

class NewCourseModal extends StatelessWidget {
  final DashboardViewModel viewModel;

  const NewCourseModal({super.key, required this.viewModel});

  @override
  Widget build(BuildContext context) {
    return Dialog(
      shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(16)),
      child: Padding(
        padding: const EdgeInsets.all(24),
        child: Column(
          mainAxisSize: MainAxisSize.min,
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Row(
              mainAxisAlignment: MainAxisAlignment.spaceBetween,
              children: [
                const Text(
                  'Create New Course',
                  style: TextStyle(fontSize: 20, fontWeight: FontWeight.bold),
                ),
                IconButton(
                  onPressed: () => viewModel.setShowNewCourseModal(false),
                  icon: const Icon(Icons.close),
                ),
              ],
            ),
            const SizedBox(height: 20),
            _buildInputField(
              'Course Title *',
              viewModel.newCourse['title'] ?? '',
                  (value) => viewModel.updateNewCourse('title', value),
            ),
            const SizedBox(height: 16),
            _buildInputField(
              'Category *',
              viewModel.newCourse['category'] ?? '',
                  (value) => viewModel.updateNewCourse('category', value),
            ),
            const SizedBox(height: 16),
            _buildTextArea(
              'Description',
              viewModel.newCourse['description'] ?? '',
                  (value) => viewModel.updateNewCourse('description', value),
            ),
            const SizedBox(height: 24),
            Row(
              children: [
                Expanded(
                  child: OutlinedButton(
                    onPressed: () => viewModel.setShowNewCourseModal(false),
                    child: const Text('Cancel'),
                  ),
                ),
                const SizedBox(width: 16),
                Expanded(
                  child: ElevatedButton(
                    onPressed: () {
                      if (viewModel.newCourse['title'].toString().isNotEmpty &&
                          viewModel.newCourse['category'].toString().isNotEmpty) {
                        viewModel.createCourse();
                      }
                    },
                    child: const Text('Create Course'),
                  ),
                ),
              ],
            ),
          ],
        ),
      ),
    );
  }

  Widget _buildInputField(String label, String value, Function(String) onChanged) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Text(
          label,
          style: const TextStyle(fontWeight: FontWeight.w600),
        ),
        const SizedBox(height: 8),
        TextField(
          onChanged: onChanged,
          decoration: InputDecoration(
            hintText: 'Ex: ${label.replaceAll(' *', '')}',
            border: OutlineInputBorder(borderRadius: BorderRadius.circular(8)),
          ),
        ),
      ],
    );
  }

  Widget _buildTextArea(String label, String value, Function(String) onChanged) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Text(
          label,
          style: const TextStyle(fontWeight: FontWeight.w600),
        ),
        const SizedBox(height: 8),
        TextField(
          onChanged: onChanged,
          maxLines: 4,
          decoration: InputDecoration(
            hintText: 'Describe what students will learn...',
            border: OutlineInputBorder(borderRadius: BorderRadius.circular(8)),
          ),
        ),
      ],
    );
  }
}

class NewLiveModal extends StatelessWidget {
  final DashboardViewModel viewModel;

  const NewLiveModal({super.key, required this.viewModel});

  @override
  Widget build(BuildContext context) {
    return Dialog(
      shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(16)),
      child: Padding(
        padding: const EdgeInsets.all(24),
        child: Column(
          mainAxisSize: MainAxisSize.min,
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Row(
              mainAxisAlignment: MainAxisAlignment.spaceBetween,
              children: [
                const Text(
                  'Schedule Live Session',
                  style: TextStyle(fontSize: 20, fontWeight: FontWeight.bold),
                ),
                IconButton(
                  onPressed: () => viewModel.setShowNewLiveModal(false),
                  icon: const Icon(Icons.close),
                ),
              ],
            ),
            const SizedBox(height: 20),
            _buildInputField(
              'Session Title *',
              viewModel.newLive['title'] ?? '',
                  (value) => viewModel.updateNewLive('title', value),
            ),
            const SizedBox(height: 16),
            _buildDateTimeField(),
            const SizedBox(height: 16),
            _buildTextArea(
              'Description',
              viewModel.newLive['description'] ?? '',
                  (value) => viewModel.updateNewLive('description', value),
            ),
            const SizedBox(height: 24),
            Row(
              children: [
                Expanded(
                  child: OutlinedButton(
                    onPressed: () => viewModel.setShowNewLiveModal(false),
                    child: const Text('Cancel'),
                  ),
                ),
                const SizedBox(width: 16),
                Expanded(
                  child: ElevatedButton(
                    onPressed: () {
                      if (viewModel.newLive['title'].toString().isNotEmpty &&
                          viewModel.newLive['datetime'].toString().isNotEmpty) {
                        viewModel.createLiveSession();
                      }
                    },
                    child: const Text('Schedule Session'),
                  ),
                ),
              ],
            ),
          ],
        ),
      ),
    );
  }

  Widget _buildInputField(String label, String value, Function(String) onChanged) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Text(
          label,
          style: const TextStyle(fontWeight: FontWeight.w600),
        ),
        const SizedBox(height: 8),
        TextField(
          onChanged: onChanged,
          decoration: InputDecoration(
            hintText: 'Ex: ${label.replaceAll(' *', '')}',
            border: OutlineInputBorder(borderRadius: BorderRadius.circular(8)),
          ),
        ),
      ],
    );
  }

  Widget _buildDateTimeField() {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        const Text(
          'Date & Time *',
          style: TextStyle(fontWeight: FontWeight.w600),
        ),
        const SizedBox(height: 8),
        TextField(
          onChanged: (value) => viewModel.updateNewLive('datetime', value),
          decoration: InputDecoration(
            hintText: 'Select date and time',
            border: OutlineInputBorder(borderRadius: BorderRadius.circular(8)),
          ),
        ),
      ],
    );
  }

  Widget _buildTextArea(String label, String value, Function(String) onChanged) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Text(
          label,
          style: const TextStyle(fontWeight: FontWeight.w600),
        ),
        const SizedBox(height: 8),
        TextField(
          onChanged: onChanged,
          maxLines: 4,
          decoration: InputDecoration(
            hintText: 'Describe what you\'ll cover in this session...',
            border: OutlineInputBorder(borderRadius: BorderRadius.circular(8)),
          ),
        ),
      ],
    );
  }
}